python/plugins/per-project-location/location/per_project_location.py

```diff
@@ -0,0 +1,78 @@
+import os
+import sys
+import platform
+import functools
+import tempfile
+import logging
+
+import ftrack_api
+from ftrack_api.entity.location import Location
+from ftrack_api.accessor.disk import DiskAccessor
+from ftrack_api.structure.standard import StandardStructure
+
+
+logger = logging.getLogger('per_project_location')
+
+
+class PerProjectLocation(Location):
+    platform = platform.system()
+
+    def _get_project_root(self, component):
+
+        target_project = component['version']['asset']['parent']['project']
+
+        # use :storage / project location: from project settings
+        target_project = self.session.query(
+            f'select disk, root from Project where id is "{target_project["id"]}"'
+        ).first()
+
+        root_folder = target_project['root']
+        if not root_folder:
+            root_folder = target_project['disk'].get(
+                platform.system().lower(), 'unix'
```

Comment:
Does this mean a Windows user will get the Unix path if Windows is not found?

Reply:
if the windows is not defined, then it'll rely on unix.
then yes, if a windows machine does not set the windows path will pick the other.
do you prefer it to error in case ?